### ISSUE TITLE
`null XOR EXISTS { () }` returns `false`

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13123]
When running the following query against an empty database:
```cypher
RETURN null XOR EXISTS { () } AS x
```

The following result is returned:
```
╒═════╕
│"x"  │
╞═════╡
│false│
└─────┘
```

I believe this result is wrong. As stated in the [docs](https://neo4j.com/docs/cypher-manual/5/syntax/operators/#query-operators-boolean), `null XOR a = null`, no matter what `a` evaluates to.

I encountered this issue when testing queries against the **Neo4j 5.6.0 community version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following queries and observe it wrongly returns `false`:
```cypher
RETURN null XOR EXISTS { () } AS x
```

### Expected behavior
The query should return `null`.

### Actual behavior
The query returns `false`.

**Comment by AzuObs** [comment]
Hi Dominic,

Thank you for reporting another issue. I agree with that something here is wrong, either with the code or the docs. I have submitted a request to the appropriate team to take a look ([private link](https://trello.com/c/KwQLfvN8)).

All the best,

**Comment by inmost-light** [comment]
@DominicWuest
Thank you for reporting. A fix has been implemented and should be available in 5.8.0.
